ckan/lib/search/__init__.py

```diff
@@ -1,318 +1,23 @@
-import logging
-import sys
-import cgitb
-import warnings
-import xml.dom.minidom
-import urllib2
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+from ckan import plugins
 
-from pylons import config
-from paste.deploy.converters import asbool
 
-import ckan.model as model
-import ckan.plugins as p
-import ckan.logic as logic
+def index(package):
+    for plugin in plugins.PluginImplementations(plugins.IIndexer):
+        return plugin.index(package)
```

Comment:
Multiple indexers would be useful to live-migrate a site to a new indexer.

1. add second indexer to end of plugins

   `warning: multiple IIndexer plugins enabled. Only the first will be used for search.`

2. search-index rebuild --only-missing to populate the new index. new and changed datasets will be updated in both indexes.

3. reorder the indexes to search with the new plugin (possibly just on one front-end)

4. if everything looks good, remove the old plugin. if not, switch back. Both plugins could be enabled for as long as it takes to make sure all your users are happy

I'd like to loop over each one when indexing, with a shallow copy because I'm sure some plugin is going to modify the dict in-place:
```python
for plugin in plugins.PluginImplementations(plugins.IIndexer):
    plugin.index(dict(package))
```